models/git/protected_branch.go

```diff
@@ -413,19 +414,52 @@ func UpdateProtectBranch(ctx context.Context, repo *repo_model.Repository, prote
 	}
 	protectBranch.ApprovalsWhitelistTeamIDs = whitelist
 
-	// Make sure protectBranch.ID is not 0 for whitelists
-	if protectBranch.ID == 0 {
-		if _, err = db.GetEngine(ctx).Insert(protectBranch); err != nil {
-			return fmt.Errorf("Insert: %v", err)
+	return db.WithTx(ctx, func(ctx context.Context) error {
```

Comment:
The `WithTx` does nothing here.

Because even if you have it, `MAX(priority)` could get the same result in 2 concurrent requests, then they will have the same final `protectBranch.Priority` values.

Reply:
there should not be two branches who are created at the same thime, and if so, the ordering rules we had before will just take place between the two.

so a same priority is totally ok.

Reply:
Yup, it is OK, but I mean "The WithTx does nothing here."

It could (should) be removed.

Reply:
63ccf8223756ca035f5130bfe3aeff8af4d04c80